### DISCUSSION TITLE
mixing exported and installed libraries

### DISCUSSION THREAD

**Discussion by eladmaimoni** [microsoft/vcpkg#24431] [Q&A]
I have a cmake project that uses a vcpkg.json manifest file to manage its dependencies.

One of the dependencies - opencv4[contrib,cuda,nonfree] is particularly heavy to compile locally. Also, it seems that binary caching doesn't work that good for this library and vcpkg ends up compiling this library from scratch even if I have multiple instances of the project on the same workstation.

I would like to avoid having to compile this library.

My thought was to use vcpkg to export this library and then integrate it to the project somehow. However, this means I have to use 2 toolchain files - one for the cmake-vcpkg integration and one for the exported packages integration. So I tried using both toolchain files. 

```
           "CMAKE_TOOLCHAIN_FILE": {
                    "value": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
                    "type": "FILEPATH"
                },
            "VCPKG_CHAINLOAD_TOOLCHAIN_FILE":  "C:/Libraries/exported_packages/scripts/buildsystems/vcpkg.cmake",
              
```  
This created some nasty errors and conflicts. I guess this is not meant to work.
So can I somehow mix exported and non-exported (installed) packages? Or is there an alternative approach which might work better?

**Comment by Osyotr** [comment]
I would look into `share/opencv4/vcpkg_abi_info.txt` first to see why binary cache doesn't work.

**Comment by bytebunny** [comment]
Hi! I just faced the same question and my current solution is to tell cmake to look for packages also in the directory of VCPKG: https://github.com/microsoft/vcpkg/discussions/38463
